timesketch/lib/testlib.py

```diff
@@ -0,0 +1,321 @@
+# limitations under the License.
+"""This module contains common test utilities for Timesketch."""
+
+from flask_testing import TestCase
+from timesketch import create_app
+from timesketch.lib import datastore
+from timesketch.models import init_db
+from timesketch.models import drop_all
+from timesketch.models import db_session
+from timesketch.models.user import User
+from timesketch.models.sketch import Sketch
+from timesketch.models.sketch import Timeline
+from timesketch.models.sketch import SearchIndex
+from timesketch.models.sketch import View
+from timesketch.models.sketch import Event
+
+
+class TestConfig(object):
+    """Config for the test environment."""
+    DEBUG = True
+    SECRET_KEY = 'testing'
+    SQLALCHEMY_DATABASE_URI = "sqlite://"
+    WTF_CSRF_ENABLED = False
+    ELASTIC_HOST = None
+    ELASTIC_PORT = None
+
+
+class MockDataStore(datastore.DataStore):
+    """A mock implementation of a Datastore."""
+    event_dict = {
+        "_index": [],
+        "_id": "adc123",
+        "_type": "plaso_event",
+        "_source": {
+            "es_index": "",
+            "es_id": "",
+            "label": "",
+            "timestamp": 1410895419859714,
+            "timestamp_desc": "",
+            "datetime": "2014-09-16T19:23:40+00:00",
+            "source_short": "",
+            "source_long": "",
+            "message": "",
+            }
+    }
+    search_result_dict = {
+        "hits": {
+            "hits": [
+                {
+                    "sort": [
+                        1410593223000
+                    ],
+                    "_type": "plaso_event",
+                    "_source": {
+                        "timestamp": 1410593222543942,
+                        "message": "Test event",
+                        "timesketch_label": [
+                            {
+                                "user_id": 1,
+                                "name": "__ts_star",
+                                "sketch_id": 1
+                            },
+                            {
+                                "user_id": 2,
+                                "name": "__ts_star",
+                                "sketch_id": 99
+                            },
+                        ],
+                        "timestamp_desc": "Content Modification Time",
+                        "datetime": "2014-09-13T07:27:03+00:00"
+                    },
+                    "_score": "null",
+                    "_index": "test",
+                    "_id": "test"
+                }
+            ],
+            "total": 1,
+            "max_score": "null"
+        },
+        "_shards": {
+            "successful": 10,
+            "failed": 0,
+            "total": 10
+        },
+        "took": 5,
+        "timed_out": False
+    }
+
+    def __init__(self, host, port):
+        """
+        Initialize the datastore. Does nothing but make sure that the
```

Comment:
I opt:
"""
Initializes the datastore object:

Args:
 host: ...
 port: ...
"""

The "Does nothing but make sure that the arguments are accepted." IMO is not relevant.